### DISCUSSION TITLE
Chia wallet password?

### DISCUSSION THREAD

**Discussion by tianxinya2021** [Chia-Network/chia-blockchain#2003] [Q&A]
Hi, I am new to Chia. I just downloaded and installed the full node. However, I do not see any password protection to my wallet. Do I miss anything? I am on win 10 computer. My win login password normally is not that secure. Any input? Thanks.

**Comment by hoffmang9** [comment]
You private key is secured by your OS Keyring on Windows.

**Reply by Bibop182** [reply]
It should be possible to enable password confirmation for sending transactions. Many users will probably agree with this.

I ask you to include these improvements in the plan for future releases, although this will require a change in RPC API.
I am ready to offer my contribution and prepare PR, if you let it.

**Reply by tianxinya2021** [reply]
What I like to see a password to protect the full node/farm/wallet, in any given OS. It is like you do not want to use only OS password to protect your banking information. I'd appreciate the implement of this security in future version. Thanks.

**Comment by tianxinya2021** [comment]
Hi, Gene:
Thanks for your reply. 

I cannot seem to find what is the function of 'connect to other peers' on the full node below block.
Is my ip address is also exposed to all the network?
Thanks.
Xinya



    On Saturday, April 17, 2021, 10:40:38 PM EDT, Gene Hoffman ***@***.***> wrote:  
 
 


You private key is secured by your OS Keyring on Windows.

**Reply by dorkmo** [reply]
yes your ip address is shared among other nodes on the network. you could likely use a VPN if you'd like more privacy

**Comment by tianxinya2021** [comment]
Thanks. I still do not understand why one would like to 'connect to other peers'. I am new here, but try to understand the key function in the full node interface. If I can connect to anyone else, does it mean anyone else can connect me without my permission?

**Comment by fullet-tortuga** [comment]
How can I backup and restore the keyring for when I restore a system backup image?
